### PR TITLE
Initial code for diagnostics

### PR DESCRIPTION
**Pull request by pcoop**
This pull-request adds an optional dependency on a cics-profile (and a hard install time pre-req for the package). It'll attempt to generate additional diagnostic output for Bundles with NODEJSAPP content.

Further tweaks will be required.

### CODE REVIEW

File: src/api/BundlePush/BundlePusher.ts [hunk 1]
```diff
@@ -527,4 +609,87 @@ export class BundlePusher {
...
+
+  private async gatherNodejsDiagnosticsFromCics(cicsSession: AbstractSession): Promise<boolean> {
+    // Issue a CMCI get to the target CICSplex
+    try {
+      const data: IResourceParms = { name: "CICSNodejsapp",
+                                     criteria: "BUNDLE=" + this.params.arguments.name,
+                                     regionName: this.params.arguments.scope,
+                                     cicsPlex: this.params.arguments.cicsplex };
+      const cmciResponse = await getResource(cicsSession, data);
+      const outputRecord = cmciResponse.response.records.cicsnodejsapp;
```
Comment (matthewpwilson) [inline src/api/BundlePush/BundlePusher.ts]: Should we guard against `cmciReponse.response` and `cmciReponse.records` being `undefined` here too?

File: package.json [hunk 2]
```diff
@@ -67,6 +68,7 @@
   },
   "peerDependencies": {
     "@zowe/cli": "^5.5.0",
+    "@zowe/cics": "^2.0.1",
```
Comment (matthewpwilson) [inline package.json]: I'm interested to see what happens if you don't have the cics plugin installed.. I'll tinker with this. Think this bit is fine for now though.

**Review by matthewpwilson — CHANGES_REQUESTED** [review]

File: package.json (hunk 2 above)
Comment (matthewpwilson) [reply package.json]: Discussed on Slack https://openmainframeproject.slack.com/archives/CC8AALGN6/p1559124222012100
Chris Boehm suggests using a regular dependency instead.

**Comment by pcoop** [comment]
I've added a region summary, something along the lines of:

CICS Regions in Scope 'CICPY004' of CICSplex 'MILLPLEX':
   Applid: CICPY004   jobname: CICPY004   jobid: STC38093

or

CICS Regions in Scope 'AOR' of CICSplex 'CAPLEX':
   Applid: CALMAS1   jobname: CALMAS1   jobid: JOB19772
   Applid: CALMAS3   jobname: CALMAS3   jobid: JOB19773

also attempted to action the previous review comments.

File: src/api/BundlePush/BundlePusher.ts [hunk 3]
```diff
@@ -644,24 +644,58 @@ export class BundlePusher {
   private async gatherNodejsDiagnosticsFromCics(cicsSession: AbstractSession): Promise<boolean> {
```
Comment (matthewpwilson) [inline src/api/BundlePush/BundlePusher.ts:644]: This method is getting quite long and does a couple of different things, suggest breaking out the code for querying the Nodejsapps and getting the Regions into separate methods.

File: src/api/BundlePush/BundlePusher.ts [hunk 4]
```diff
@@ -644,24 +644,58 @@ export class BundlePusher {
...
+          cmciRegionResponse.response === undefined ||
+          cmciRegionResponse.response.records === undefined ||
+          cmciRegionResponse.response.records.cicsregion === undefined) {
+        throw new Error("CICSRegion CMCI output record not found.");
       }
+      const outputRegionRecords = cmciRegionResponse.response.records.cicsregion;
+      let msg = "CICS Regions in Scope '" + this.params.arguments.scope + "' of CICSplex '" + this.params.arguments.cicsplex + "':\n";
+      this.issueMessage(msg);
 
       // We may have an array of records if there was more than one NODEJSAPP in the bundle
```
Comment (matthewpwilson) [inline src/api/BundlePush/BundlePusher.ts:663]: I think this comment should refer to regions

**Review by matthewpwilson — CHANGES_REQUESTED** [review]

File: src/api/BundlePush/BundlePusher.ts (hunk 4 above)
Comment (pcoop) [reply src/api/BundlePush/BundlePusher.ts:663]: I was thinking the region stuff might want separating entirely as that could be useful for any Bundle whether it has Nodejs is content or not. I figured I'd return to that in another update though as it involves a bit more restructuring.

**Review by matthewpwilson — APPROVED** [review]

**Comment by pcoop** [comment]
This should address the main concern in issue #33.

**Pull request merged**